src/fetchtastic/downloader.py

```
@@ -1192,44 +1338,68 @@ def extract_version(dir_name: str) -> str:
+
+    def _safe_get_timestamp(commit_hash):
         try:
-            if compare_versions(dir_version, latest_release_version) > 0:
-                repo_prerelease_dirs.append(dir_name)
-        except (InvalidVersion, ValueError, TypeError) as exc:
+            return (
+                get_commit_timestamp("meshtastic", "firmware", commit_hash)
+                if commit_hash
+                else None
+            )
+        except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)